src/lib/server/orders.ts

```diff
@@ -659,6 +659,19 @@ export async function createOrder(
 		throw error(400, "You can't use free payment method on this order");
 	}
 
+	items.forEach((item) => {
+		if (item.chosenVariations && item.product.variations) {
+			const variationNamesInDB = [...new Set(item.product.variations.map((vari) => vari.name))];
+			const chosenVariationNames = Object.keys(item.chosenVariations);
+			const allVariationsChosen =
+				variationNamesInDB.length === chosenVariationNames.length &&
+				variationNamesInDB.every((name) => chosenVariationNames.includes(name));
+
+			if (!allVariationsChosen) {
+				throw error(400, 'error matching on variations choice');
+			}
+		}
+	});
```

Comment:
We should re-compute `customPrice` when there are variations.

In case the variation price changes between adding to cart and creating the order

Reply:
On a functional behavior PoV I'm more into "price displayed when adding to cart should stay to the end" (except for currency variation, that are fixed once a cart becomes an order through checkout).
I let you decide @coyotte508 but for me it's not so much of an issue if we add a product 1 with variation A to cart, the price change, and we still have the same price on /checkout.
It makes me think more about product price change while a cart exist and is not an order yet, but, for variation price ,I don't really care for now that we refresh it.